src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -283,68 +404,52 @@ export default class CanvasController {
                 round(this.canvas.size.width / 2 + this.canvas.panX),
                 round(this.canvas.size.height / 2 + this.canvas.panY)
             );
-            // TODO: this is incorrect usage of a ternary. ternaries should be used for a ssignment not function calls.
-            // draw airspace border
-            AirportController.airport_get().airspace
-                ? this.canvas_draw_airspace_border(cc)
-                : this.canvas_draw_ctr(cc);
+
+            if (AirportController.airport_get().airspace) {
+                this.canvas_draw_airspace_border(cc);
+            } else {
+                this.canvas_draw_ctr(cc);
+            }
 
             this.canvas_draw_range_rings(cc);
             cc.restore();
 
             // Special markings for ENGM point merge
             if (AirportController.airport_get().icao === 'ENGM') {
                 cc.save();
-                cc.translate(
-                    calculateMiddle(this.canvas.size.width),
-                    calculateMiddle(this.canvas.size.height)
-                );
+                cc.translate(middleWidth, middleHeight);
                 this.canvas_draw_engm_range_rings(cc);
                 cc.restore();
             }
 
             // Compass
             cc.font = 'bold 10px monoOne, monospace';
 
-            if (this.canvas.dirty || fading || true) {
-                cc.save();
-                cc.translate(
-                    calculateMiddle(this.canvas.size.width),
-                    calculateMiddle(this.canvas.size.height)
-                );
+            cc.save();
+            cc.translate(middleWidth, middleHeight);
 
-                this.canvas_draw_compass(cc);
-                cc.restore();
-            }
+            this.canvas_draw_compass(cc);
+            cc.restore();
 
             cc.font = BASE_CANVAS_FONT;
 
-            if (this.canvas.dirty || this.canvas_should_draw() || true) {
+            if (this.canvas_should_draw() || true) {
                 cc.save();
                 cc.globalAlpha = alpha;
-                cc.translate(
-                    calculateMiddle(this.canvas.size.width),
-                    calculateMiddle(this.canvas.size.height)
-                );
+                cc.translate(middleWidth, middleHeight);
                 this.canvas_draw_all_aircraft(cc);
```

Comment:
Generally speaking, it's quite silly we've been passing the canvas context around between every method involved in the draw. I assume you have plans to do so, but it would make a lot of sense to have the 2d context as a property of the CanvasController itself so we can instead call the methods without any arguments, and they can just do `this._ctx` or something to get it.

Reply:
yes it is a little awkward, however, it means we only need to get the `context` once.  And, we are actually storing it too.  `this.canvas.contexts[CANVAS_NAME].context` is where that lives.  for now, it's just easier grabbing it once and passing it around.

This will likely change, though, as I make my way through these changes.